rfcs/0007-repartitioning.md

```diff
@@ -0,0 +1,139 @@
+# Repartitioning Primitives
+
+## Motivation and Problem Statement
+
+Currently, SlateDB does not include a built-in partitioning mechanism. To scale horizontally, users must implement their own partitioning strategy and manage each partition as a separate SlateDB instance. This design is consistent with other LSM-tree–based embedded databases, where partitioning is often left to higher-level orchestration.
+
+While creating static partitions is relatively straightforward, changing the number of partitions over time (repartitioning) is more complex. Existing embedded databases like RocksDB and Badger do not provide convenient primitives to enable a simple and performant repartitioning workflow in user applications.
```

Comment:
i suppose that iceberg or paimon might be good references for the partitioning scheme, like: 

https://iceberg.apache.org/docs/1.4.0/partitioning/

also, it supports a smooth evolution of repartitioning: 

https://iceberg.apache.org/docs/1.4.0/evolution/#partition-evolution